### PR TITLE
Add more random Avatar generators (DiceBear, Robot) and make it configurable

### PR DESCRIPTION
**Pull request by 6543**
DiceBear: **https://codeberg.org/Codeberg/avatars**
Robot: **https://github.com/lafriks/go-avatars**
None: use default images
 
* [x] POC
* [ ] Settings via ini
* [ ] more tests :)

### CODE REVIEW

File: modules/avatar/avatar.go [hunk 1]
```diff
@@ -8,44 +8,39 @@ import (
...
 
-	"github.com/issue9/identicon"
 	"github.com/nfnt/resize"
 	"github.com/oliamb/cutter"
 )
 
 // AvatarSize returns avatar's size
 const AvatarSize = 290
 
+type RandomAvatarGenerator interface {
```
Comment (6543) [inline modules/avatar/avatar.go]: should we do some sort of interface & register or just import and use sub packages?

**Comment by wxiaoguang** [comment]
Can such feature be optional? I just worry if we introduce too many packages, the size of Gitea binary would become too big, sooner and sooner.

And it seems that the random avatar can use the "gravatar" mechanism, just point the gravatar webserver to the  DiceBear avatar server.

**Comment by 6543** [comment]
the problem ... as is what you propose would be best to solve via #2222 whitch to not exist :/

**Comment by wxiaoguang** [comment]
Maybe #14799 can be easily fixed by a left-right mirror random avatar.

**Comment by wxiaoguang** [comment]
Please have a look at:

* https://github.com/go-gitea/gitea/pull/17707

**Comment by Gusted** [comment]
Should this be added as a second option or be closed since #17707 is merged?

**Comment by 6543** [comment]
I'm for a second option since it does not add mouch to the size ... but asap we have a plugin system we should make both to plugins and fallback to the "default" avatar

**Comment by 6543** [comment]
and about the svg lib: I'll also use that for other features ... like the request to be able to upload svg as profile image etc ...

**Comment by wxiaoguang** [comment]
ps: the random generated DiceBear avatars have beard/braid, I think it's not suitable for a general purpose avatar generator before we know user's gender or preference.

**Comment by lafriks** [comment]
We could also use my library https://github.com/lafriks/go-avatars (it does generate png avatars also)

**Comment by wxiaoguang** [comment]
Quote my question:

> Can such feature be optional? I just worry if we introduce too many packages, the size of Gitea binary would become too big, sooner and sooner.
> And it seems that the random avatar can use the "gravatar" mechanism

To avoid code bloat, do we really want to introduce new avatar generators? Could the avatars provided by external gravatar servers instead of compile code into Gitea?

**Comment by 6543** [comment]
`TAGS="bindata sqlite sqlite_unlock_notify" make build -j1`

before (0d05bdbea8a4ffdd37e7268db75f7a1a8fdcadfb): 84M (85384)
at 0a0b147513e5e5f56cc6475ec406ab49b497a0c1: 84M (85900)
at 8c6bed3f1956f9e95c8c81fc4223afb6851b23b9: 85M (86196)

so I think we can remove [monsterid](https://github.com/go-gitea/gitea/pull/17701/commits/309c5e1f6e311c9239497c69bb7a8e4428521a4b) and [wavatars](https://github.com/go-gitea/gitea/pull/17701/commits/8c6bed3f1956f9e95c8c81fc4223afb6851b23b9)

but the rest should be good to go - and yes you can use gravatar like service but it will not be that fine granular & easy to do for selfhost admins

**Comment by wxiaoguang** [comment]
To avoid bloating Gitea's code base, let users use their own avatar service (like self-maintained gavatar) to provide their customized avatars?


After #17707 , nobody complains about `swastika avatars #14799`

**Comment by wxiaoguang** [comment]
I am cleaning stale/pending PRs, according to the discussion and vote above, I think this PR could be closed.

**Pull request closed without merging**